src/storage/src/mut_records.rs

```diff
@@ -24,16 +24,36 @@ use crate::records::FileRecords;
 
 pub const MESSAGE_LOG_EXTENSION: &str = "log";
 
+pub enum MutFileRecordsFlushPolicy {
+    NoFlush,
+    EveryWrite,
+    CountWrites { n_writes: u32 },
+}
+
 /// Can append new batch to file
 pub struct MutFileRecords {
     base_offset: Offset,
     item_last_offset_delta: Size,
     f_sink: BoundedFileSink,
+    flush_policy: MutFileRecordsFlushPolicy,
+    flush_write_count: u32, // only used for CountWrite policy
```

Comment:
can you change to `write_count`?  I think this is useful for general metrics.

Reply:
sure no problem

Reply:
Making this either a saturating or wrapping counter is pretty straightforward it looks like